zcash_client_sqlite/src/wallet.rs

```diff
@@ -1964,6 +1972,90 @@ pub(crate) fn store_transaction_to_be_sent<P: consensus::Parameters>(
+    // at present for fully-transparent transactions, because any transaction with a shielded
+    // component will be detected via ordinary chain scanning and/or nullifier checking.
+    if !detectable_via_scanning {
+        queue_tx_retrieval(
+            wdb.conn.0,
+            std::iter::once(sent_tx.tx().txid()),
+            TxQueryType::Status,
+            None,
+        )?;
+    }
+
+    Ok(())
+}
+
+pub(crate) fn set_transaction_status(
+    conn: &rusqlite::Transaction,
+    txid: TxId,
+    status: TransactionStatus,
+) -> Result<(), SqliteClientError> {
+    match status {
+        TransactionStatus::TxidNotRecognized | TransactionStatus::NotInMainChain => {
+            // If the transaction is now expired, remove it from the retrieval queue.
+            if let Some(chain_tip) = chain_tip_height(conn)? {
+                let mut stmt = conn.prepare_cached(
+                    "SELECT expiry_height FROM transactions
+                     WHERE txid = :txid",
+                )?;
+
+                let expiry_height = stmt
+                    .query_row(named_params![":txid": txid.as_ref()], |row| {
+                        row.get::<_, Option<u32>>(0)
+                            .map(|opt| opt.map(BlockHeight::from))
+                    })
+                    .optional()?
+                    .flatten();
+
+                // If the transaction's expiry height + the confirmation depth is less than the
+                // chain tip height, we know it to be un-mineable and we can remove the reference
+                // to it from the transaction retrieval queue.
+                if expiry_height
+                    .iter()
+                    .any(|expiry| *expiry + VERIFY_LOOKAHEAD < chain_tip)
+                {
+                    conn.execute(
+                        "DELETE FROM tx_retrieval_queue WHERE txid = :txid",
+                        named_params![
+                            ":txid": txid.as_ref()
+                        ],
+                    )?;
+                }
+            }
+        }
+        TransactionStatus::Mined(height) => {
+            // The transaction has been mined, so we can set its mined height, associate it with
+            // the appropriate block, and remove it from the retrieval queue.
+            let sql_args = named_params![
+                ":txid": txid.as_ref(),
+                ":height": u32::from(height)
+            ];
+
+            conn.execute(
+                "UPDATE transactions
+                 SET mined_height = :height
+                 WHERE txid = :txid",
+                sql_args,
+            )?;
+
+            conn.execute(
+                "UPDATE transactions
+                 SET block = blocks.height
+                 FROM blocks
+                 WHERE txid = :txid
+                 AND blocks.height = :height",
+                sql_args,
+            )?;
```

Comment:
I believe this may fix https://github.com/zcash/librustzcash/pull/1473/files#diff-14c1d11caabdb0c9a159f3a3ae58987177180bb9739872c7ea93cfd4952e9d37R667-R700 . If it does, please remove the workaround from that test.

Reply:
At present, `set_transaction_status` is not called by any path in existing tests; it is explicitly introduced as a method for the wallet app to invoke once it has satisfied a transaction data request. It may be possible to update the test to simulate that interaction.

Reply:
I will check that.

Reply:
Filed #1485.